classes/user/form/IdentityForm.php

```diff
@@ -121,9 +122,20 @@ public function execute(...$functionArgs)
         $request = Application::get()->getRequest();
         $user = $request->getUser();
 
-        $user->setGivenName($this->getData('givenName'), null);
-        $user->setFamilyName($this->getData('familyName'), null);
-        $user->setPreferredPublicName($this->getData('preferredPublicName'), null);
+
+        // Request to delete ORCID token is handle separately from other form field updates
+        if($this->getData('removeOrcidId') === 'true') {
+            $clonedUser = clone $user;
+            $user->setOrcid(null);
+            $user->setOrcidVerified(false);
+
+            OrcidManager::removeOrcidAccessToken($user);
+            dispatch(new RevokeOrcidToken($request->getContext(), $clonedUser));
```

Comment:
Same as comment above about moving the job dispatching into `OrcidManager::removeOrcidAccessToken()`.